### PR TITLE
Exams Krykota Davyd IM-33

### PR DESCRIPTION
**Pull request by DavidkJun**

### CODE REVIEW

**Comment by elaiviaien** [comment]
I have reviewed the changes. I agree with all of the commits

**Comment by yur-ochka** [comment]
I have reviewed the changes and totally agree with them.

**Comment by namanica** [comment]
good job has been done!

**Comment by Kiryaoo** [comment]
Good job with code, all clean and organized

**Comment by nikama-s** [comment]
i have reviewed all commits and agree with everything

**Comment by nevsetakpohano** [comment]
I have reviewed all commits, and you've done such a great job! Especially, I like that you have been so detailed in your commit descriptions

**Comment by D-Vasylchenko** [comment]
I looked at the work, very well done, I agree with all the commits

**Comment by D-Vasylchenko** [comment]
I reviewed the work, very well done, I agree with all obligations

**Comment by D-Vasylchenko** [comment]
I reviewed this work, very well done, I agree with all obligations

File: Tasks/duplicate.js [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
-duplicate = (value, N) => {
-  if (N <= 0) return [];
-  else {
-    res = [];
-    for (let i = 0; i < N; i++) {
-      res[i] = value;
+const duplicate = (valueToRepeat, timesToRepeat) => {
+  if (timesToRepeat <= 0) {
+    return [];
+  } else {
```
Comment (tshemsedinov) [inline Tasks/duplicate.js:7]: We do not need else here having return above

File: Tasks/duplicate.js [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+    const duplicates = [];
```
Comment (tshemsedinov) [inline Tasks/duplicate.js:8]: Next step is new Array and fill

File: Tasks/month.js [hunk 3]
```diff
@@ -1,26 +1,15 @@
...
-  'nov',
-  'dec',
-];
-
-Month = (s) => {
-  l = Months.length;
-  for (let i = 0; i < l; i++) {
-    if (s.toLowerCase().startsWith(Months[i])) return i + 1;
+const findMonth = (monthToFind) => {
+  const months = ['jan','feb', 'mar','apr','may','jun','jul','aug','sep','oct','nov', 'dec',];
```
[the remark this answers is not in the document]
Comment (tshemsedinov) [reply Tasks/month.js:5]: Move this out of function

File: Tasks/month.js [hunk 4]
```diff
@@ -1,26 +1,15 @@
...
+  const numberOfMonths = months.length;
+  for (let i = 0; i < numberOfMonths; i++) {
+    if (monthToFind.toLowerCase().startsWith(months[i])) {
+      return i + 1;
```
Comment (tshemsedinov) [inline Tasks/month.js:9]: 1. Extract loop invariant: `monthToFind.toLowerCase();`
2. Use for..of here
3. `if (lowered.startsWith(month)) return i + 1;`

File: Tasks/password.js [hunk 5]
```diff
@@ -1,13 +1,16 @@
...
+  const maxLength = alphabet.length;
+  let password = '';
 
-let GeneratePassword = (alphabet, length) => {
-  const MAX = alphabet.length;
-  key = '';
   for (let i = 0; i < length; i++) {
-    Index = Math.floor(Math.random() * MAX);
-    key = key + alphabet[Index];
+    const index = Math.floor(Math.random() * maxLength);
```
Comment (tshemsedinov) [inline Tasks/password.js:10]: Would be great to decompose: extract `random` function

File: Tasks/sample.js [hunk 6]
```diff
@@ -1,8 +1,10 @@
...
+'use strict';
 
-const SAMPLE = (arr) => {
-  arr = arr[Math.floor(Math.random() * arr.length)];
-  return arr;
+const getSample = (arr) => {
+let sampleIndex = 0;
+sampleIndex = Math.floor(Math.random() * arr.length);
+  return arr[sampleIndex];
 };
```
Comment (tshemsedinov) [inline Tasks/sample.js:8]: Terrible

File: Tasks/types.js [hunk 7]
```diff
@@ -1,18 +1,18 @@
...
+const typesCount =  (arrayOfTypes) => {
+  const types = {
     number: 0,
     string: 0,
     boolean: 0,
   };
-  for (i of s) {
-    const t = typeof i;
-    types_[t]++;
+  for (const type of arrayOfTypes) {
```
Comment (tshemsedinov) [inline Tasks/types.js:10]: Type in worst naming here, better use `value`

Comment (DavidkJun) [reply Tasks/types.js]: reworked that

File: Tasks/month.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (DavidkJun) [reply Tasks/month.js]: reworked that

File: Tasks/day.js [hunk 8]
```diff
@@ -1,15 +1,15 @@
...
+'use strict';
 
-_parse_day_ = (s) => {
-  let i;
-  for (i = 0; i < D.length; i++) {
-    if (s.startsWith(D[i].toLowerCase())) {
+const getDayNumber = (dayToFind) => {
+  const days = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+  for (let i = 0; i < days.length; i++) {
+    if (dayToFind.startsWith(days[i].toLowerCase())) {
```
[the remark this answers is not in the document]
Comment (DavidkJun) [reply Tasks/day.js]: reworked that

File: Tasks/duplicate.js (hunk 2 above)
Comment (DavidkJun) [reply Tasks/duplicate.js]: reworked that

File: Tasks/month.js (hunk 4 above)
Comment (DavidkJun) [reply Tasks/month.js]: reworked that

File: Tasks/password.js (hunk 5 above)
Comment (DavidkJun) [reply Tasks/password.js]: reworked that

File: Tasks/sample.js (hunk 6 above)
Comment (DavidkJun) [reply Tasks/sample.js]: reworked that

**Pull request closed without merging**